### PR TITLE
TDR-3276 - Add ADR for metadata input validation

### PR DESCRIPTION
**Pull request by vimleshtna**

### CODE REVIEW

File: architecture-decision-records/0030-metadata-input-validation.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# 4. Metadata input validation
```
Comment (TomJKing) [inline architecture-decision-records/0030-metadata-input-validation.md:1]: Number should match the ADR number, ie: `... 30. Metadata input ...`

File: architecture-decision-records/0030-metadata-input-validation.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+**Date**: 2023-08-14
+
+## Context
+
+TDR has a feature to add additional metadata by submitting the form. When the user submits the form, we validate the input data
```
Comment (TomJKing) [inline architecture-decision-records/0030-metadata-input-validation.md:7]: Reword slightly to: `... by submitting an HTML form. ...`

File: architecture-decision-records/0030-metadata-input-validation.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+and show an error on the page if the input data is incorrect or if everything is correct then we add the data into the database.
+Now, we will be adding another feature to add metadata by uploading a CSV file.
+So, again we need to validate the CSV file and if everything is correct then add the data into the database otherwise show an error.
+
+To avoid multiple validation, we thought to have a common validation for form and CSV file both.
```
Comment (TomJKing) [inline architecture-decision-records/0030-metadata-input-validation.md:12]: Reword slightly: `... validation HTML form and CSV file.`

File: architecture-decision-records/0030-metadata-input-validation.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+## Decision
+
+Use a validation library.
+
+We found that a library will be very useful for form and CSV file validation. We will send input data (form and CSV data) and metadata criteria,
+it will then check data against the metadata criteria, and if the data is invalid then will return error codes to generate error messages
+as per source.
+
+The other benefit is that we can use this library to validate the input data in [Consignment API][consignment-api] too as we are validating on API while
```
Comment (TomJKing) [inline architecture-decision-records/0030-metadata-input-validation.md:24]: Reword slightly to: `... validating in the API while ...`

**Review by TomJKing — COMMENTED** [review]
Looks good. Thanks for writing this up. Just some minor comments.

File: architecture-decision-records/0030-metadata-input-validation.md (hunk 1 above)
Comment (vimleshtna) [reply architecture-decision-records/0030-metadata-input-validation.md]: Good catch, I have updated it, thanks.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**